### DISCUSSION TITLE
skynet连接mysql执行sql慢,但是手动执行很快

### DISCUSSION THREAD

**Discussion by cheungfeifu** [cloudwu/skynet#1747] [Q&A]
代码如下：执行sql很慢，大约5s左右，但是数据库慢查询设置阈值1s，确实没有捕获到。大家知道时间在哪里浪费了嘛？
skynet版本为：版本为 v1.3.0 (2019-11-19）
function CMD.query(sql)
	local startTime = skynet.time()

	local res = db:query(sql)

	local timeUsed = skynet.time() - startTime
	-- 记录大于100毫秒的请求
	if timeUsed>0.1 then
		local time = (math.floor(timeUsed*1000000)/1000)
		local msg = "time:"..time.." sql:"..sql
		skynet.send("LOG", "lua", "log", msg, "db_slow", "")
	end

	return res
end

**Comment by cloudwu** [comment]
建议先模仿 https://github.com/cloudwu/skynet/blob/master/test/testmysql.lua 硬编码你的慢查询单独测试。

**Reply by ericzhanchina** [reply]
云风，有空看看我这个问题吧，谢谢，又有点不一样。
https://github.com/cloudwu/skynet/discussions/1742

**Comment by hanxi** [comment]
https://plantegg.github.io/2020/11/17/MySQL%E7%BA%BF%E7%A8%8B%E6%B1%A0%E5%AF%BC%E8%87%B4%E7%9A%84%E5%BB%B6%E6%97%B6%E5%8D%A1%E9%A1%BF%E6%8E%92%E6%9F%A5/

可以尝试排查一下是不是mysql设置的问题。
